### PR TITLE
add comments & accumulate bug fix for TOSS

### PR DESCRIPTION
**Pull request by liuyu85cn**
1. add comments & adjust some log level 
2. accumulate bug fix for TOSS
    2.1 one space may be affected by another space. when recover prime edges. 
    2.2 when call kvstore->get(), if leaderLeaseInvalid, it will report leader_change, change it to leader_lease_failed.
    2.3 print all execution trace in ChainProcessor::finish(), make much more grep friendly. 
    2.4 some function of memory lock is hard to understand (forceLock, forceUnlock) , change to "setAutoUnlock(bool)"

#### What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### What problem(s) does this PR solve?
Issue(s) number: 

Description:

#### Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/storage/transaction/ChainAddEdgesLocalProcessor.cpp [hunk 1]
```diff
@@ -363,9 +288,31 @@ cpp2::AddEdgesRequest ChainAddEdgesLocalProcessor::reverseRequest(
...
+            << ", rcRemote_=" << apache::thrift::util::enumNameSafe(rcRemote_)
+            << ", rcCommit_=" << apache::thrift::util::enumNameSafe(rcCommit_);
+  }
+  do {
+    if (rcPrepare_ != Code::SUCCEEDED) {
+      break;  // nothing written, no need to recover.
+    }
+
+    if (rcCommit_ != Code::SUCCEEDED) {
+      reportFailed(ResumeType::RESUME_CHAIN);
```
Comment (critical27) [inline src/storage/transaction/ChainAddEdgesLocalProcessor.cpp:304]: I see `addPrime` is called in `reportFailed`, why we need to do it again?

**Review by critical27 — COMMENTED** [review]
Any special reason to use `rcPrepare_`, `rcRemote_` and `rcCommit_`, it actually make the parameter in `processRemote` and `processLocal` is useless, and the chain is broken. Is it mainly for we could set the result in `finish` method?

**Comment by liuyu85cn** [comment]
> Any special reason to use `rcPrepare_`, `rcRemote_` and `rcCommit_`, it actually make the parameter in `processRemote` and `processLocal` is useless, and the chain is broken. Is it mainly for we could set the result in `finish` method?

use this style because previously, when pass a code in the chain, it's hard to tell where an error happens. 
for example if processLocal() get an error code "leader change", 
it can't determined if this error happened in prepareLocal() or processRemote().
actually, I use some other member to identify this error ("primeInserted" for example.)

It is the same reason why set result in finish(). Because I could get a clear view of what this processor has done.
The previous version, will report to transaction manager at the moment some error happens. 
This is really error prone because I need to carefully check not to report a same error twice.

File: src/storage/transaction/ChainAddEdgesLocalProcessor.cpp (hunk 1 above)
Comment (liuyu85cn) [reply src/storage/transaction/ChainAddEdgesLocalProcessor.cpp:304]: In my opinion,  "addPrime()" is really hard to understand.
"reportFailed" is just a self explain/comment function .

**Review by critical27 — APPROVED** [review]

**Comment by critical27** [comment]
@kikimo you could do a final test later

File: src/storage/test/CMakeLists.txt [hunk 2]
```diff
@@ -754,20 +754,20 @@ nebula_add_executable(
...
-    OBJECTS
-        ${storage_test_deps}
-    LIBRARIES
-        ${ROCKSDB_LIBRARIES}
-        ${THRIFT_LIBRARIES}
-        ${PROXYGEN_LIBRARIES}
-        wangle
-        gtest
-)
+# nebula_add_executable(
```
Comment (panda-sheep) [inline src/storage/test/CMakeLists.txt:757]: Why comment out? 
If don't need it, can delete it or add a comment

**Review by critical27 — DISMISSED** [review]

**Comment by kikimo** [comment]
> @kikimo you could do a final test later

I'll do it this afternoon

File: src/storage/test/CMakeLists.txt (hunk 2 above)
Comment (liuyu85cn) [reply src/storage/test/CMakeLists.txt]: ok, delete it for short

**Review by panda-sheep — DISMISSED** [review]
Good job!

**Comment by kikimo** [comment]
> > @kikimo you could do a final test later
> 
> I'll do it this afternoon

Checked, no problem.

**Review by kikimo — DISMISSED** [review]
LGTM

**Review by kikimo — APPROVED** [review]
LGTM

**Review by critical27 — APPROVED** [review]

**Pull request merged**